site/pages/admin/social-media-accounts.page.tsx

```diff
@@ -1,51 +1,180 @@
+                            grant_type: "refresh_token",
+                            refresh_token: token.value ?? "",
+                        }),
+                        headers: {
+                            "Content-Type": "application/x-www-form-urlencoded",
+                            Authorization: authToken,
+                        },
+                    });
+                    if (resp.ok) {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                        const tokenResult: HootsuiteToken = await resp.json();
+                        const keys = await getParametersByPath(`/social/${orgId}/hootsuite`);
+
+                        if (!keys || (refreshTokens && keys.Parameters?.length === 0)) {
+                            throw new Error("Refresh token is required to fetch dropdown data");
+                        }
+                        const key: string =
+                            keys.Parameters?.find((rt) => rt.Name?.includes(`${token.userId}`))?.Name || "";
+                        if (!key) {
+                            throw new Error("Refresh token is required to fetch dropdown data");
+                        }
+
+                        await putParameter(key, tokenResult.refresh_token ?? "", "SecureString", true);
+                        const userDetailsResponse = await fetch(`${HOOTSUITE_URL}v1/me`, {
+                            method: "GET",
+                            headers: {
+                                Authorization: `Bearer ${tokenResult.access_token ?? ""}`,
+                            },
+                        });
+                        if (userDetailsResponse.ok) {
+                            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                            const userDetails: HootsuiteMe = await userDetailsResponse.json();
+                            const userInfo = userDetails.data || {};
+
+                            const extraInfo = {
+                                ...userInfo,
+                                accountType: token.accountType || "",
+                                addedBy: token.addedBy || "",
+                                expiresIn: "Never",
+                            } as SocialMediaAccountsSchema[0];
+
+                            const socialProfilesResponse = await fetch(`${HOOTSUITE_URL}v1/socialProfiles`, {
+                                method: "GET",
+                                headers: {
+                                    Authorization: `Bearer ${tokenResult.access_token ?? ""}`,
+                                },
+                            });
+                            if (socialProfilesResponse.ok) {
+                                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                                const socialProfiles: HootsuiteSocialProfiles = await socialProfilesResponse.json();
```

Comment:
As this is an external response - it should be parsed using zod

Reply:
done